src/contracts/middleware/BLSPublicKeyCompendium.sol

```diff
@@ -20,49 +19,44 @@ contract BLSPublicKeyCompendium is IBLSPublicKeyCompendium {
 
     // EVENTS
     /// @notice Emitted when `operator` registers with the public key `pk`.
-    event NewPubkeyRegistration(address operator, BN254.G1Point pubkeyG1, BN254.G2Point pubkeyG2);
+    event NewPubkeyRegistration(address indexed operator, BN254.G1Point pubkeyG1, BN254.G2Point pubkeyG2);
 
     /**
      * @notice Called by an operator to register themselves as the owner of a BLS public key and reveal their G1 and G2 public key.
-     * @param s is the field element of the operator's Schnorr signature
-     * @param rPoint is the group element of the operator's Schnorr signature
-     * @param pubkeyG1 is the the G1 pubkey of the operator
-     * @param pubkeyG2 is the G2 with the same private key as the pubkeyG1
+     * @param signedMessageHash is the registration message hash signed by the private key of the operator
+     * @param pubkeyG1 is the corresponding G1 public key of the operator 
+     * @param pubkeyG2 is the corresponding G2 public key of the operator
      */
-    function registerBLSPublicKey(uint256 s, BN254.G1Point memory rPoint, BN254.G1Point memory pubkeyG1, BN254.G2Point memory pubkeyG2) external {
-        // calculate -g1
-        BN254.G1Point memory negGeneratorG1 = BN254.negate(BN254.G1Point({X: 1, Y: 2}));
-        // verify a Schnorr signature (s, R) of pubkeyG1
-        // calculate s*-g1 + (R + H(msg.sender, P, R)*P) = 0
-        // which is the Schnorr signature verification equation
-        BN254.G1Point memory shouldBeZero = 
-            BN254.plus(
-                BN254.scalar_mul(negGeneratorG1, s),
-                BN254.plus(
-                    rPoint,
-                    BN254.scalar_mul(
-                        pubkeyG1,
-                        uint256(keccak256(abi.encodePacked(msg.sender, pubkeyG1.X, pubkeyG1.Y, rPoint.X, rPoint.Y))) % BN254.FR_MODULUS
-                    )
-                )
-            );
+    function registerBLSPublicKey(BN254.G1Point memory signedMessageHash, BN254.G1Point memory pubkeyG1, BN254.G2Point memory pubkeyG2) external {
+        // H(m) 
+        BN254.G1Point memory messageHash = BN254.hashToG1(keccak256(abi.encodePacked(
+            msg.sender, 
+            block.chainid, 
+            "EigenLayer_BN254_Pubkey_Registration"
+        )));
```

Comment:
IMO the message hash should also include `address(this)`.
I also highly recommend adding a public getter function for the message hash, for convenience.